### DISCUSSION TITLE
counting calls

### DISCUSSION THREAD

**Discussion by JoostvanBruggen** [Mattk70/Chirpity-Electron#170] [Ideas]
I use Chirpity a lot... Already I find it the best app in years that I use.
I use it (as many others) for nocmig.

Nocmig is about migration and counting birds or the number of calls, right?

Well, as all apps for BIRD-ID's the machine uses the 3-second time to ID a birdspecies. But in this timeframe a Redwing, for instance, can call two times or a flock passes by.

Would it be possible to ad a whole new possibility to count every call the app heard? This way many NocMiggers would be helped.

Thanks,

Joost van Bruggen, Holland

**Comment by Mattk70** [comment]
Thanks!

You have hit upon a limitation of the kind of AI model used for detecting the species. The class of model used (a Convolutional Neural Network) is the most accurate for identification. However, it can only identify the presence of a call or calls in the time window. Other classes of network, such as Transformers, might be better at counting call / song events, but with the amount of labelled data available, struggle to achieve anything like the accuracy - rendering them pretty useless. 

A combination of approaches may be the best solution with current technology. That is, use the existing AI model to find periods with calls, then use a different technique to count the calls in the periods they were detected. It would be a considerable piece of work, and there's no guarantee (in fact it's quite unlikely) the results would approach the level of accuracy achieved by a manual count.

Meanwhile, Chirpity at least allows you to enter a count, but you have to do this for each detection with multiple calls to get an accurate total. Oh, and you'll also need to adjust for occasions where one single call is detected twice!

For me, it is sufficient that Chirpity can give a clear indication of a heavy passage vs. a light one for a species, and also to know that the number of records reported by Chirpity can be compared across nights, since the counting technique is consistent and repeatable, if the settings used are also the same.
